### PR TITLE
Login with handle

### LINKED ISSUE #511: login as handle
allows user to login with a "handle" (name, not email) of their choice and that is what is stored in leaderboard, contest, results.

### PR DESCRIPTION
**Pull request by TraceyOnim**
closes #511 

Peek 2021-09-15 11-12

### CODE REVIEW

File: lib/quadquizaminos_web/views/layout_view.ex [hunk 1]
```diff
@@ -49,6 +49,9 @@ defmodule QuadquizaminosWeb.LayoutView do
           <a class = "button" href="<%= Routes.auth_path(QuadquizaminosWeb.Endpoint, :request, "linkedin") %>">
           Sign in with LinkedIn
           </a>
+          <a class = "button" href="<%= Routes.session_path(QuadquizaminosWeb.Endpoint, :new) %>">
```
Comment (sigu) [inline lib/quadquizaminos_web/views/layout_view.ex:52]: Needs to be the first option in the drop down list

File: lib/quadquizaminos_web/controller/session_controller.ex [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    render(conn, "new.html")
+  end
+
+  def create(conn, %{"session" => %{"name" => name}}) do
+    case Accounts.find_or_create(name) do
+      {:ok, user} ->
+        conn
+        |> put_flash(:info, "Successfully authenticated.")
+        |> assign(:current_user, user)
+        |> put_session(:uid, user.name)
```
Comment (sigu) [inline lib/quadquizaminos_web/controller/session_controller.ex:15]: The UID should be the  user.uid, not the user name. If this means that we creat a uid same as the name then thats okay

File: lib/quadquizaminos_web/views/layout_view.ex (hunk 1 above)
Comment (TraceyOnim) [reply lib/quadquizaminos_web/views/layout_view.ex]: worked on it

File: lib/quadquizaminos_web/controller/session_controller.ex (hunk 2 above)
Comment (TraceyOnim) [reply lib/quadquizaminos_web/controller/session_controller.ex]: worked on it

**Pull request merged**